kairon/shared/cognition/processor.py

```diff
@@ -112,6 +112,112 @@ def list_cognition_schema(self, bot: Text):
             final_data['collection_name'] = collection
             yield final_data
 
+    def delete_collection_data(self, collection_id: str, bot: Text):
+        try:
+            collection = CollectionData.objects(bot=bot, id=collection_id).get()
+            collection.delete()
+        except DoesNotExist:
+            raise AppException("Collection Data does not exists!")
+
+    @staticmethod
+    def validate_collection_payload(collection_name, is_secure, data):
+        if not collection_name:
+            raise AppException("collection name is empty")
+
+        if not isinstance(is_secure, list):
+            raise AppException("is_secure should be list of keys")
+
+        if is_secure:
+            if not data or not isinstance(data, dict):
+                raise AppException("Invalid value for data")
+
+    def save_collection_data(self, payload: Dict, user: Text, bot: Text):
+        collection_name = payload.get("collection_name", None)
+        data = payload.get('data')
+        is_secure = payload.get('is_secure')
+        CognitionDataProcessor.validate_collection_payload(collection_name, is_secure, data)
+
+        if Utility.is_exist(CollectionData, bot=bot, collection_name__iexact=collection_name,
```

Comment:
collection name can be duplicate, as we will maintaing data against single type(collection_name)